component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -763,6 +788,15 @@ private void redirectToEmailAddressReqPage(HttpServletResponse response, Authent
             String emailAddressReqPage = getEmailAddressRequestPage(context, emailOTPParameters);
             try {
                 String url = getRedirectURL(emailAddressReqPage, queryParams);
+                if (isEmailUpdateFailed(context)) {
+                    url = url + EmailOTPAuthenticatorConstants.RETRY_PARAMS;
```

Comment:
use FrameworkUtils.appendQueryParamToURL

Reply:
Fixed this issue